uid2_client/encryption.py

```diff
@@ -227,17 +233,13 @@ def _decrypt_token_v3(token_bytes, keys, domain_name, client_type, now, token_ve
         return DecryptedToken(DecryptionStatus.DOMAIN_NAME_CHECK_FAILED, None, None, site_id, site_key.site_id,
                           keys.get_identity_scope(), identity_type, token_version, is_client_side_generated, expires)
 
-    established_ms = int.from_bytes(site_payload[20:28], 'big')
     established = dt.datetime.fromtimestamp(established_ms / 1000.0, tz=timezone.utc)
-    # refreshed_ms 28:36
+    generated = dt.datetime.fromtimestamp(generated_ms / 1000.0, tz=timezone.utc)
 
-    if not _token_has_valid_lifetime(keys, client_type, established, expires, now):
+    if not _token_has_valid_lifetime(keys, client_type, generated, expires, now):
         return DecryptedToken(DecryptionStatus.INVALID_TOKEN_LIFETIME, None, established, site_id, site_key.site_id,
```

Comment:
Probably it would be good to change the `established` in the error to `generated`

Reply:
as above